bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexTeamsHandler.java

```diff
@@ -0,0 +1,461 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WebexTeamsHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Tom Deckers - Initial contribution
+ */
+@NonNullByDefault
+public class WebexTeamsHandler extends BaseThingHandler implements AccessTokenRefreshListener {
+
+    private final Logger logger = LoggerFactory.getLogger(WebexTeamsHandler.class);
+
+    // Object to synchronize refresh on
+    private final Object refreshSynchronization = new Object();
+
+    private @NonNullByDefault({}) WebexTeamsConfiguration config;
+
+    private final OAuthFactory oAuthFactory;
+    private final HttpClient httpClient;
+    private @Nullable WebexTeamsApi client;
+
+    private @Nullable OAuthClientService authService;
+
+    private boolean configured = false; // is the handler instance properly configured?
+    private volatile boolean active; // is the handler instance active?
+    String accountType = ""; // bot or person?
+
+    private @Nullable Future<?> refreshFuture;
+
+    public WebexTeamsHandler(Thing thing, OAuthFactory oAuthFactory, HttpClient httpClient) {
+        super(thing);
+        this.oAuthFactory = oAuthFactory;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No commands supported on any channel
+    }
+
+    // creates list of available Actions
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singletonList(WebexTeamsActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing thing {}", this.getThing().getUID());
+        active = true;
+        config = getConfigAs(WebexTeamsConfiguration.class);
+
+        final String token = config.token;
+        final String clientId = config.clientId;
+        final String clientSecret = config.clientSecret;
+
+        if (!token.isBlank()) { // For bots
+            logger.debug("I think I'm a bot.");
+            try {
+                createBotOAuthClientService(config);
+            } catch (WebexTeamsException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorNotAuth");
+                return;
+            }
+        } else if (!clientId.isBlank()) { // For integrations
+            logger.debug("I think I'm a person.");
+            if (clientSecret.isBlank()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorNoSecret");
+                return;
+            }
+            createIntegrationOAuthClientService(config);
+        } else { // If no bot or integration credentials, go offline
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorTokenOrId");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Just move it after the next check with authService

Reply:
Gotcha.. done.